File: Data Structures & Algorithms/longest-strictly-increasing-or-strictly-decreasing-subarray/submission-3.rs
impl Solution {
    pub fn longest_monotonic_subarray(nums: Vec<i32>) -> i32 {
        let n = nums.len();
        let mut res = 1;

        for i in 0..n - 1 {
            let mut cur_len = 1;
            for j in i + 1..n {
                if nums[j] == nums[j - 1]
                    || (nums[i] < nums[i + 1]) != (nums[j - 1] < nums[j])
                {
                    break;
                }
                cur_len += 1;
            }
            res = res.max(cur_len);
        }

        res
    }
}

File: Data Structures & Algorithms/can-place-flowers/submission-3.java
class Solution {
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        int len = flowerbed.length;
        int[] res = new int[len + 2];

        for(int i = 0; i < len; i++) {
            res[i + 1] = flowerbed[i];
        }
        
        for(int i = 0; i < len; i++) {
            System.out.println(res[i]);
        }

        for(int i = 1; i <= len; i++) {
            if(res[i-1] == 0 && res[i] == 0 && res[i+1] == 0) {
                res[i] = 1;
                n--;
            }
        }
        return n == 0;
    }
}